pulsar-broker/src/main/java/org/apache/pulsar/broker/service/schema/SchemaRegistryService.java

```diff
@@ -42,19 +44,29 @@
         return checkers;
     }
 
-    static SchemaRegistryService create(SchemaStorage schemaStorage, Set<String> schemaRegistryCompatibilityCheckers) {
+    static SchemaRegistryService create(String schemaRegistryName, SchemaStorage schemaStorage,
+          Set<String> schemaRegistryCompatibilityCheckers) {
         if (schemaStorage != null) {
             try {
                 Map<SchemaType, SchemaCompatibilityCheck> checkers = getCheckers(schemaRegistryCompatibilityCheckers);
                 checkers.put(SchemaType.KEY_VALUE, new KeyValueSchemaCompatibilityCheck(checkers));
-                return SchemaRegistryServiceWithSchemaDataValidator.of(
-                        new SchemaRegistryServiceImpl(schemaStorage, checkers));
+
+                if (schemaRegistryName == null) {
+                return SchemaRegistryServiceWithSchemaDataValidator
+                      .of(new SchemaRegistryServiceImpl(schemaStorage, checkers));
+                } else {
+                    return (SchemaRegistryService) Class.forName(schemaRegistryName)
```

Comment:
Better to have a unit test to make sure the configuration works well.

Reply:
the default case is already tested in the existing test cases. i'll add some more test cases to explicitly test this part.

Reply:
have updated this logic. we are using a default value in the config and continuing to use the schema validator wrapper class.
the reasoning was that any custom schema registry implementation would need to comply with existing usages of SchemaRegistryService interface, including pulsar-admin apis and the client connection flows. the SchemaRegistryServiceWithSchemaDataValidator wrapper class ensures that any input schemas are validated via avro/json/protobuf-native parser before the compatibility checks and schema storage interactions are applied.